phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -927,7 +928,16 @@ private void addCoprocessors(byte[] tableName, TableDescriptorBuilder builder, P
             boolean isViewIndex = TRUE_BYTES_AS_STRING
                     .equals(tableProps.get(MetaDataUtil.IS_VIEW_INDEX_TABLE_PROP_NAME));
 
-            if ((tableType == PTableType.INDEX && !isTransactional) || isViewIndex) {
+            boolean isViewBaseTransactional = false;
+            if (!isTransactional && isViewIndex) {
+                if (tableProps.containsKey(TRANSACTIONAL) &&
+                        Boolean.TRUE.equals(tableProps.get(TRANSACTIONAL))) {
```

Comment:
Is "Boolean.TRUE.equals(tableProps.getOrDefault(TRANSACTIONAL, FALSE))" more efficient?